prospero-common/src/main/java/org/wildfly/prospero/api/ArtifactUtils.java

```diff
@@ -31,25 +34,39 @@ public static int compareVersion(Artifact first, Artifact other) {
         return new ComparableVersion(first.getVersion()).compareTo(new ComparableVersion(other.getVersion()));
     }
 
-    public static ChannelManifestCoordinate manifestFromString(String urlGavOrPath) {
+    public static ChannelManifestCoordinate manifestCoordFromString(String urlGavOrPath) {
+        return coordinateFromString(urlGavOrPath, ChannelManifestCoordinate.class);
+    }
+
+    public static ChannelCoordinate channelCoordFromString(String urlGavOrPath) {
+        return coordinateFromString(urlGavOrPath, ChannelCoordinate.class);
+    }
+
+    private static <T extends ChannelMetadataCoordinate> T coordinateFromString(String urlGavOrPath, Class<T> clazz) {
         try {
-            URL url = new URL(urlGavOrPath);
-            return new ChannelManifestCoordinate(url);
-        } catch (MalformedURLException e) {
-            if (isValidCoordinate(urlGavOrPath)) {
-                try {
-                    return ChannelManifestCoordinate.create(null, urlGavOrPath);
-                } catch (MalformedURLException ex) {
-                    throw new RuntimeException(ex);
+            try {
+                URL url = new URL(urlGavOrPath);
+                return clazz.getDeclaredConstructor(URL.class).newInstance(url);
+            } catch (MalformedURLException e) {
+                if (isValidCoordinate(urlGavOrPath)) {
+                    String[] gav = urlGavOrPath.split(":");
+                    if (gav.length == 2) {
+                        return clazz.getDeclaredConstructor(String.class, String.class).newInstance(gav[0], gav[1]);
+                    } else if (gav.length == 3) {
+                        return clazz.getDeclaredConstructor(String.class, String.class, String.class)
+                                .newInstance(gav[0], gav[1], gav[2]);
+                    }
                 }
-            } else {
                 // assume the string is a path
                 try {
-                    return new ChannelManifestCoordinate(Paths.get(urlGavOrPath).toAbsolutePath().toUri().toURL());
+                    return clazz.getDeclaredConstructor(URL.class)
```

Comment:
Maybe add a check that the file exists?

Reply:
I wouldn't do that here. We don't check URLs and GAVs for existence either.